packages/rest/src/body-parsers/body-parser.json.ts

```diff
@@ -36,6 +38,9 @@ export class JsonBodyParser implements BodyParser {
 
   async parse(request: Request): Promise<RequestBody> {
     let body = await invokeBodyParserMiddleware(this.jsonParser, request);
+
+    if (!_.isPlainObject(body)) throw RestHttpErrors.invalidRequestBody();
```

Comment:
I think this is not the right place for rejecting array values for object-only bodies. What if the request body was sent as urlencoded formdata (or XML in the future when we add support for parsing XML request)? We also don't know what's the allowed body schema at this point.

I would expect this additional validation to happen inside [request-body.validator.ts](https://github.com/strongloop/loopback-next/blob/cf97cd9a9b8ea1ff7334de7ffb7bb870580486e2/packages/rest/src/validation/request-body.validator.ts).

Note that we already have logic to validate that non-body parameters are valid objects, see here:

https://github.com/strongloop/loopback-next/blob/cf97cd9a9b8ea1ff7334de7ffb7bb870580486e2/packages/rest/src/coercion/coerce-parameter.ts#L161-L174

Can we perhaps reuse this logic for request bodies too?

TBH, I am surprised that AJV is not handling this edge case for us and allows array values when the schema describes an object 🤔